src/python_testing/TC_IDM_4_2.py

```diff
@@ -385,41 +427,67 @@ async def test_TC_IDM_4_2(self):
         )
 
         # Verify that the subscription is activated between CR1 and DUT
-        asserts.assert_true(sub_cr1_provided_dvf.subscriptionId, "Subscription not activated")
+        asserts.assert_true(sub_cr1_step7.subscriptionId, "Subscription not activated")
 
-        sub_cr1_provided_dvf.Shutdown()
+        sub_cr1_step7.Shutdown()
 
         # *** Step 8 ***
-        self.print_step(8, "CR1 sends a subscription request action for an attribute and sets the MinIntervalFloor value to be same as MaxIntervalCeiling. Activate the Subscription between CR1 and DUT. Modify the attribute which has been subscribed to on the DUT.")
+        self.print_step(8, "CR1 sends a subscription request action for an attribute and sets the MinIntervalFloor to min_interval_floor_sec and MaxIntervalCeiling to 10. Activate the Subscription between CR1 and DUT and record the time when the priming ReportDataMessage is received as t_report_sec. Save the returned MaxInterval from the SubscribeResponseMessage as max_interval_sec.")
 
         # Subscribe to attribute
-        same_min_max_interval_sec = 3
+        min_interval_floor_sec = 3
+        max_interval_ceiling_sec = 10
         sub_cr1_update_value = await CR1.ReadAttribute(
             nodeid=self.dut_node_id,
             attributes=node_label_attr_path,
-            reportInterval=(same_min_max_interval_sec, same_min_max_interval_sec),
+            reportInterval=(min_interval_floor_sec, max_interval_ceiling_sec),
             keepSubscriptions=False
         )
 
-        # Modify attribute value
+        # Record the time when the priming ReportDataMessage is received
+        t_report_sec = time.time()
+
+        # *** Step 9 ***
+        self.print_step(9, "CR1 modifies the attribute which has been subscribed to on the DUT and waits for an incoming ReportDataMessage")
+
+        # Saving the returned MaxInterval from the SubscribeResponseMessage
+        sub_cr1_step9_intervals = sub_cr1_update_value.GetReportingIntervalsSeconds()
+        min_interval_floor_sec, max_interval_sec = sub_cr1_step9_intervals
+
+        # Get subscription timeout
+        subscription_timeout_sec = sub_cr1_update_value.GetSubscriptionTimeoutMs() / 1000
+
+        # Set Attribute Update Callback
+        node_label_queue = queue.Queue()
```

Comment:
can you move this to be a member of the AttributeChangeCallback? I think that'll make it easier to use. Then you can just make a wait function on that class and we can re-use this whole things quite easily.

ie instead of wait_for_attribute_report, you could then do node_label_update_cb.wait_for_report()

Reply:
Done